src/Loader/StrictPoLoader.php

```diff
@@ -0,0 +1,349 @@
+use Gettext\Translations;
+
+/**
+ * Class to load a PO file.
+ */
+final class StrictPoLoader extends Loader
+{
+    /** @var Translations */
+    private $translations;
+    /** @var Translation */
+    private $translation;
+    private $data;
+    private $position;
+    private $inPreviousComment;
+
+    public function loadString(string $string, Translations $translations = null): Translations
+    {
+        $this->data = $string;
+        $this->position = 0;
+        $this->translations = parent::loadString($string, $translations);
+        for ($this->newEntry(); $this->getChar() !== null; $this->newEntry()) {
+            while ($this->readComment());
+            $this->readContext();
+            $this->readId();
+            if ($this->readPlural()) {
+                for ($isRequired = true; $this->readPluralTranslation($isRequired); $isRequired = false);
+            } else {
+                $this->readTranslation();
+            }
+            $this->saveEntry();
+        }
+        $this->processHeader();
+        return $this->translations;
+    }
+
+    private function newEntry(): void
+    {
+        $this->inPreviousComment = false;
+        $this->translation = $this->createTranslation(null, '');
+    }
+
+    private function saveEntry(): void
+    {
+        $this->translations->add($this->translation);
+    }
+
+    private function readDisabledComment(): bool
+    {
+        return $this->translation->isDisabled() && $this->readString('#~');
+    }
+
+    private function readPreviousTranslationComment(): bool
+    {
+        return $this->inPreviousComment && $this->readString('#|');
+    }
+
+    private function readWhiteSpace(): bool
+    {
+        $position = $this->position;
+        while ((ctype_space($this->getChar() ?? '') && $this->nextChar())
+            || $this->readDisabledComment()
+            || $this->readPreviousTranslationComment());
+        return $position !== $this->position;
+    }
+
+    private function readString(string $word): bool
+    {
+        return substr($this->data, $this->position, strlen($word)) === $word
+            ? !!($this->position += strlen($word))
+            : false;
+    }
+
+    private function readChar(string $char): bool
+    {
+        return $this->getChar() === $char
+            ? !!++$this->position
+            : false;
+    }
+
+    private function nextChar(): ?string
+    {
+        $char = $this->getChar();
+        if ($char !== null) {
+            ++$this->position;
+        }
+        return $char;
+    }
+
+    private function getChar(): ?string
+    {
+        return $this->data[$this->position] ?? null;
+    }
+
+    private function readNumber(): string
+    {
+        for ($data = ''; ctype_digit($this->getChar() ?? ''); $data .= $this->nextChar());
+        return $data;
+    }
+
+    private function readCommentString(): string
+    {
+        for ($data = ''; ($this->getChar() ?? "\n") !== "\n"; $data .= $this->nextChar());
```

Comment:
Here I think it might be interesting to add a `\r` to support different line breaks (a user using `\r\n` will probably end up with the `\r` being added to the `$data`).